File: typescript/tree.ts
export class TreeNode {
    val: number
    left: TreeNode | null
    right: TreeNode | null
    constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
        this.val = (val === undefined ? 0 : val)
        this.left = (left === undefined ? null : left)
        this.right = (right === undefined ? null : right)
    }
}

export const buildTree = (vals: (number | null)[]): TreeNode | null => {
    if (!vals.length) return null;

    const root = new TreeNode(vals[0]!);
    let nodes: TreeNode[] = [root];
    vals = vals.slice(1);
    while (vals.length) {
        const node = nodes.shift()!;

        if (vals[0] !== null && vals[0] !== undefined) {
            node.left = new TreeNode(vals[0]!);
            nodes.push(node.left);
        }
        if (vals[1] !== null && vals[1] !== undefined) {
            node.right = new TreeNode(vals[1]!);
            nodes.push(node.right);
        }
        vals = vals.slice(2);
    }

    return root;
};

File: python/tree.py
from typing import List, Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def buildTree(vals: List[int | None]) -> Optional[TreeNode]:
    if not vals or vals[0] is None:
        return None
    root = TreeNode(vals[0])
    nodes = [root]
    vals = vals[1:]
    while vals:
        node = nodes.pop(0)
        if vals[0] is not None:
            node.left = TreeNode(vals[0])
            nodes.append(node.left)
        if vals[1] is not None:
            node.right = TreeNode(vals[1])
            nodes.append(node.right)
        vals = vals[2:]
    return root


def compareTree(t1: Optional[TreeNode], t2: Optional[TreeNode]) -> bool:
    if not t1 and not t2:
        return True
    if not t1 or not t2:
        return False
    if t1.val != t2.val:
        return False
    return compareTree(t1.left, t2.left) and compareTree(t1.right, t2.right)
